packages/vite/src/node/build.ts

```diff
@@ -319,7 +319,7 @@ export function resolveBuildPlugins(config: ResolvedConfig): {
     ],
     post: [
       buildImportAnalysisPlugin(config),
-      buildEsbuildPlugin(config),
+      ...(config.esbuild === false ? [] : [buildEsbuildPlugin(config)]),
```

Comment:
I think it'll be more consistent if we do 
```suggestion
      ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
```

so it's similar to https://github.com/vitejs/vite/blob/8b21029f9dcd1a63e8b58f3d54cef12c53ebef53/packages/vite/src/node/plugins/index.ts#L54

and the lines below